demos/Cargo.toml

```diff
@@ -1,123 +1,123 @@
 [package]
-name = "smol-examples"
+name = "smol-demos"
 version = "0.0.0"
 authors = ["Stjepan Glavina <stjepang@gmail.com>"]
 edition = "2018"
 publish = false
 
-[target.'cfg(windows)'.dev-dependencies]
+[target.'cfg(windows)'.dependencies]
 uds_windows = "0.1.4"
 
-[target.'cfg(target_os = "linux")'.dev-dependencies]
-inotify = { version = "0.8.2", default-features = false }
-nix = "0.17.0"
+[target.'cfg(target_os = "linux")'.dependencies]
+inotify = { version = "0.8", default-features = false }
+nix = "0.17"
 timerfd = "1.1.1"
 
-[dev-dependencies]
-anyhow = "1.0.28"
```

Comment:
I'm not necessarily opposed to this change, but why remove minor and patch part of version numbers here?

With this change version numbers do look cleaner, but my worry is that we could be relying on APIs that e.g. exist in version "1.1.0", but don't exist in "1.0.0", in which case specifying version "1" is not quite correct.

Reply:
In my experience, many Rust crates are fine with introducing backwards-incompatible changes within minor versions, so this could potentially break everything accidentally.

Reply:
Am thinking that should be ok, since these are demos? Am also thinking that, if one wanted to be extra sure, they'd use the `=1.2.3` anyways? Also, is it not the case that cargo ignores the `2` in `v1.2.3` when you `cargo update` anyways. Sorry, the nuances of cargo versioning aren't super clear to me.

Reply:
It should be no big deal since these are demos :) But why change the versions we had before? 

Specifying `dep = "1"` is same as `dep = "1.0.0"`. But if we're using APIs that were introduced only in version 1.2, then we should require `dep = "1.2"`.

Reply:
Oh, I thought `dep = "1"` meant all of `dep = "1.*.*"`. If that's case, removing them removes the awkwardness of having a `dep = "1.2.3"` be valid also for `dep = "1.3.0"`.

Reply:
Just to clarify, `dep = "1"` means "any version between 1.0.0 and 2.0.0". If you're using features that were introduced in 1.5, then you need to specify `dep = "1.5"` because you need a version between 1.5.0 and 2.0.0.

Reply:
Oh, didn't know that. Let me revert.